### DISCUSSION TITLE
Handle custom URL schemes (with python transformations)

### DISCUSSION THREAD

**Discussion by dzervas** [qutebrowser/qutebrowser#8282] [Q&A]
I want to make qutebrowser able to handle custom schemes. for example `tg://hello` should result in `https://t.me/hello`. How can I intercept any URL passed to the CLI of qutebrowser?

**Comment by The-Compiler** [comment]
Can you elaborate on what the problem is you're trying to solve with this? Again #30 is probably the right place for this (the "Custom URL schemes" part), but I'm a bit confused on what the goal is here.

**Reply by dzervas** [reply]
Many native apps use custom schemes so that you can go to a specific place - e.g. to open a specific profile. For example uses the `tg://` scheme so that the OS opens the telegram desktop app during handling. I want qutebrowser to be "mimic" the native app, get the URL and handle it by opening a URL

Cool, I'll wait for the plugins support, thanks!

**Reply by toofar** [reply]
I think you should be able to do this already with a wrapper script. If you register a script for the desired schemes then in the script transform the URL and pass it to qutebrowser (or back to xdg-open or whatever for the default https handler to pick up).
I might do that for slack so I don't have to click the "open in your browser link" manually.
